### PR TITLE
Allow for natural coordinates in input file for point values

### PR DESCRIPTION
**Pull request by anne-glerum**

### CODE REVIEW

File: source/postprocess/point_values.cc [hunk 1]
```diff
@@ -194,11 +211,22 @@ namespace aspect
                                        ">, but this does not correspond to a list of numbers with "
                                        "as many coordinates as you run your simulation in."));
 
-              Point<dim> point;
+              //Point<dim> point;
```
Comment (MFraters) [inline source/postprocess/point_values.cc]: please remove this.

File: source/postprocess/point_values.cc [hunk 2]
```diff
@@ -166,6 +178,11 @@ namespace aspect
                             "The list of points at which the solution should be evaluated. "
                             "Points need to be separated by semicolons, and coordinates of "
                             "each point need to be separated by commas.");
+          prm.declare_entry("Use natural coordinates", "false",
+                            Patterns::Bool (),
+                            "Whether or not the Evaluation points are specified in "
+                            "the natural coordinates of the geometry model, e.g. "
+                            "spherical coordinates for the spherical shell.");
```
Comment (MFraters) [inline source/postprocess/point_values.cc]: Maybe add specifically the geometry models which are supported and are not supported at the moment.

File: include/aspect/postprocess/point_values.h [hunk 3]
```diff
@@ -85,8 +85,24 @@ namespace aspect
...
-        std::vector<Point<dim> >                                       evaluation_points;
+        /**
+         * Vector of arrays representing the points where the solution is to be evaluated.
+         */
+        std::vector<std_cxx11::array<double,dim> >                     evaluation_points;
+        /**
+         * Vector of Points representing the points where the solution is to be evaluated
+         * that can be used by VectorTools.
+         */
+        std::vector<Point<dim> >                                       evaluation_points_cartesian;
```
Comment (MFraters) [inline include/aspect/postprocess/point_values.h]: It seems strange to me that you need to store the data twice. Do you really need both or can't you store it once and convert when needed?

File: tests/point_value_natural_coordinates.prm [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    set Thermal conductivity          = 1e-6 # default: 4.7
+    set Thermal expansion coefficient = 2e-5
+    set Viscosity                     = 1e22    # default: 5e24
+  end
+end
+
+
+subsection Mesh refinement
+  set Initial adaptive refinement        = 0
+  set Initial global refinement          = 4
```
Comment (MFraters) [inline tests/point_value_natural_coordinates.prm]: Do you really need such a refinement for this test?

File: source/postprocess/point_values.cc [hunk 5]
```diff
@@ -200,6 +194,9 @@ namespace aspect
         {
           const std::vector<std::string> point_list
             = Utilities::split_string_list(prm.get("Evaluation points"), ';');
+
+          std::vector<std_cxx11::array<double,dim> > evaluation_points;
```
Comment (MFraters) [inline source/postprocess/point_values.cc]: you can now change the std_cxx11:: to std:: :)

File: source/postprocess/point_values.cc [hunk 6]
```diff
@@ -194,11 +208,31 @@ namespace aspect
                                        ">, but this does not correspond to a list of numbers with "
                                        "as many coordinates as you run your simulation in."));
 
-              Point<dim> point;
+              std_cxx11::array<double,dim> point;
```
Comment (MFraters) [inline source/postprocess/point_values.cc]: same here

File: source/postprocess/point_values.cc [hunk 7]
```diff
@@ -194,11 +208,31 @@ namespace aspect
...
+
+          use_natural_coordinates = prm.get_bool("Use natural coordinates");
+
+          if (use_natural_coordinates)
+            AssertThrow (dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()) == 0 &&
+                         dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model()) == 0,
+                         ExcMessage ("This postprocessor can not be used if the geometry "
+                                     "is a sphere or spherical shell, because these geometries have not implemented natural coordinates."));
+
+          // Convert the vector of - maybe natural - coordinate arrays to a
```
Comment (MFraters) [inline source/postprocess/point_values.cc]: what you you mean with maybe natural?

**Comment by anne-glerum** [comment]
@MFraters happy?

**Pull request merged**